src/Codeception/Lib/Connector/Yii2.php

```diff
@@ -148,35 +127,42 @@ public function doRequest($request)
         $yiiRequest->setQueryParams($_GET);
 
         try {
+            /*
+             * This is basically equivalent to $app->run() without sending the response.
+             * Sending the response is problematic because it tries to send headers.
+             */
             $app->trigger($app::EVENT_BEFORE_REQUEST);
-
-            $app->handleRequest($yiiRequest)->send();
-
+            $response = $app->handleRequest($yiiRequest);
             $app->trigger($app::EVENT_AFTER_REQUEST);
+            codecept_debug($response->isSent);
+            $response->send();
```

Comment:
Now it is using correct response object to get headers and cookies, response status code. This was an issue for me in previous versions, when you returned another Response object from controller. Thanks for the fix, it works!